### DISCUSSION TITLE
Mobile 2 Mobile Communication Using Petra Wallet (React Native/Expo)

### DISCUSSION THREAD

**Discussion by HarjjotSinghh** [aptos-labs/aptos-developer-discussions#379] [Questions]
### Discord user ID

h4r.

### Describe your question in detail.

Hey guys, I was trying out the [Mobile 2 Mobile Communication](https://petra.app/docs/mobile-deeplinks#mobile-2-mobile-communication) in my React Native/Expo project.

I am running into an issue while connecting my app with the Petra wallet. Upon Clicking on the `connect` button, I get redirected to the Petra app where it asks me to approve or reject the request. Upon approving the request, it does nothing and does not redirect me back to the app. If I manually redirect back to the app, the connect button is still active and I am not connected. I guess that the `DAPP_LINK_BASE`, which in my case is `avocodos:///api/v1` does not exist (I have `"scheme": "avocodos"` already setup in my `app.json` config file. 

Here is my entire `ConnectWallet.jsx` screen/component: https://codefile.io/f/qVabjtnTdc

Any help will be much appreciated.

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

N/A

**Comment by Alexzander-Stone** [comment]
Hey there, @HarjjotSinghh!

From looking at your code, it seems like the logic is correct. So your intuition about it relating to the constants you've defined, like `DAPP_LINK_BASE`, seems promising. 

Does your `AndroidManifest.xml` and `Info.plist` include the required definitions for deep linking?
Here are the [android](https://github.com/aptos-labs/mobile2mobile-example/blob/e81072946f7e66ed22320093718fa725a7612112/android/app/src/main/AndroidManifest.xml#L13) and [ios](https://github.com/aptos-labs/mobile2mobile-example/blob/e81072946f7e66ed22320093718fa725a7612112/ios/Mobile2MobileExample/Info.plist#L55) examples from, [Mobile2MobileExample](https://github.com/aptos-labs/mobile2mobile-example).
